image.h

```diff
@@ -20,18 +20,35 @@ class Image {
 	uint16_t pixelIndex(uint16_t x, uint16_t y) {
 		return x * m_columns + y;
 	}
-	// Change the name of this function... Try a more specific name instead of 'get'
-	uint8_t getPixel(uint16_t x, uint16_t y) {
+	uint8_t getPixelAtIndex(uint16_t x, uint16_t y) {
 		return m_pixels[pixelIndex(x, y)];
 	}
-	void scanPixels(std::function<void(uint8_t, uint16_t, uint16_t)> takePixel) {
+	void scanPixels(std::function<void(uint8_t, uint16_t, uint16_t, uint16_t)> takePixel) {
 		for (int x = 0; x < m_rows; x++) {
 			for (int y = 0; y < m_columns; y++) {
-				takePixel(getPixel(x, y), x, y);
+				takePixel(getPixelAtIndex(x, y), x, y, pixelIndex(x,y));
 			}
 		}
 	}
 	const uint16_t m_rows;
 	const uint16_t m_columns;
 	uint8_t* m_pixels; // max 1k x 1k image
 };
+
+class RawImage {
+public:
+	RawImage(std::shared_ptr<Image> rawImage) {
+		m_rawImage = std::make_shared<Image>(512, 512, [](uint8_t* pixels) {});
+	}
+private:
+	std::shared_ptr<Image> m_rawImage;
```

Comment:
What's the purpose of keeping this private? Cannot access anything at all, after initialization?

Reply:
I had to change to enable usage of Raw and BrightenedImage as types